NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -1,140 +1,328 @@
+    Parameters
+    ----------
+    observer : value
+        observer as in the hdf5 file object, e.g. corsika['CoREAS']['observers'].values()[i]
+    zenith : float
+        zenith angle in radians from corsika file, e.g. use get_angles()
+    azimuth : float
+        azimuth angle in radians from corsika file, e.g. use get_angles()
+    magnetic_field_vector : np.array (3)
+        magnetic field vector from corsika file, e.g. use get_angles()
+  
+    Returns
+    -------
+    obs_positions_vxB_vxvxB: np.array (3)
+        observer positions in (vxB, vxvxB) shower plane
+    
+    """
+    cs = coordinatesystems.cstrafo(zenith, azimuth, magnetic_field_vector)
 
-    return weights
+    obs_positions_geo = convert_obs_positions_to_nuradio_on_ground(observer, zenith, azimuth, magnetic_field_vector)
+
+    # transforms the coreas observer positions into the vxB, vxvxB shower plane
+    obs_positions_vxB_vxvxB = cs.transform_to_vxB_vxvxB(obs_positions_geo).T
+
+    return obs_positions_vxB_vxvxB
+
+
+def read_CORSIKA7(input_file, declination=None):
+    """
+    this function reads the corsika hdf5 file and returns a sim_station with all relevent information from the file
+
+    Parameters
+    ----------
+    input_file : string
+        path to the corsika hdf5 file
+
+    Returns
+    -------
+    evt : event object
+        event object containing the corsika information
+    """
+    if (declination is None):
+        declination = 0
+        logger.warning(
+            "No declination given, assuming 0 degree. This might need to incorrect electric field polarizations.")
+    corsika = h5py.File(input_file, "r")
+    sampling_rate = 1. / (corsika['CoREAS'].attrs['TimeResolution'] * units.second)
+    zenith, azimuth, magnetic_field_vector = get_angles(corsika, declination)
+
+    sim_shower = NuRadioReco.framework.radio_shower.RadioShower()
+    sim_shower.set_parameter(shp.zenith, zenith)
+    sim_shower.set_parameter(shp.azimuth, azimuth)
+    sim_shower.set_parameter(shp.magnetic_field_vector, magnetic_field_vector)
+    energy = corsika['inputs'].attrs["ERANGE"][0] * units.GeV
+    sim_shower.set_parameter(shp.energy, energy)
+    sim_shower.set_parameter(shp.shower_maximum, corsika['CoREAS'].attrs['DepthOfShowerMaximum'] * units.g / units.cm2)
+    sim_shower.set_parameter(shp.refractive_index_at_ground, corsika['CoREAS'].attrs["GroundLevelRefractiveIndex"])
+    sim_shower.set_parameter(shp.magnetic_field_rotation,
+                             corsika['CoREAS'].attrs["RotationAngleForMagfieldDeclination"] * units.degree)
+    sim_shower.set_parameter(shp.distance_shower_maximum_geometric,
+                             corsika['CoREAS'].attrs["DistanceOfShowerMaximum"] * units.cm)
+
+    sim_shower.set_parameter(shp.observation_level, corsika["inputs"].attrs["OBSLEV"] * units.cm)
+    sim_shower.set_parameter(shp.primary_particle, corsika["inputs"].attrs["PRMPAR"])
+    if 'ATMOD' in corsika['inputs'].attrs.keys():
```

Comment:
Is this every not True?

Reply:
Oh yes it can. If you do not specify this keyword corsika will use the standard US atmosphere `1`. Furthermore if you simulated the atmosphere with the GDAS tool this keyword is not set (or does not need to be set) IIRC.